### ISSUE TITLE
Unable to use psst on Eduroam

### ISSUE THREAD

**Issue by TrakJohnson** [jpochyla/psst#557]
**Describe the bug**
When using Psst while connected to an [eduroam](https://eduroam.org/) network, I get `Error: connection refused (os error 111)`. This applies regardless of if I was logged in before or not; anything that requires sending requests to spotify fails.

I would be happy to add logs but no clue as to where to find them.

**To Reproduce**
- Connect to eduroam network
- Open Psst (try to login / play music if already logged in)

**Expected behavior**
Being able to use Psst; I never had software specifically fail on eduroam networks and the official spotify client works fine.

**Screenshots**
![Screenshot From 2024-11-15 09-42-10](https://github.com/user-attachments/assets/523df548-471f-4c34-bf4e-304ddb2b2574)
![Screenshot From 2024-11-15 10-03-21](https://github.com/user-attachments/assets/985b69cd-4fc1-446b-a235-acb502f2be3a)

**Environment**
 - OS: Fedora Linux 41 (Workstation Edition) x84 
 - Kernel: Linux 6.11.5-300.fc41.x86_64
 - Version: commit hash 5fb8918

**Additional context**
I have checked that simply changing to another wifi solves everything. Thanks :)

**Comment by TrakJohnson** [comment]
Thank you for your replies, I currently don't have access to the network to test anything but will report back as soon as I can!

**Comment by raiden-e** [comment]
I've noticed the same. However, using a VPN seems to fix this 😆

**Comment by SO9010** [comment]
Yeah i suspect its the uni policy, mine allows it, perhaps using a proxy would also work!

**Comment by TrakJohnson** [comment]
@SO9010 pinging seems to be blocked on the university network (none of the transmitted packets received, even for google.com). I can however "ping" using `telnet` or `arping` and these work fine for both domains:

```
$ arping open.spotify.com                                                          ~
ARPING 199.232.171.42 from 192.168.155.77 wlp1s0
Unicast reply from 199.232.171.42 [28:6F:7F:FB:A6:7F]  6.044ms
Unicast reply from 199.232.171.42 [28:6F:7F:FB:A6:7F]  14.270ms
Unicast reply from 199.232.171.42 [28:6F:7F:FB:A6:7F]  10.015ms
Unicast reply from 199.232.171.42 [28:6F:7F:FB:A6:7F]  43.879ms
```

```
$ telnet api.spotify.com 443                                                       ~
Trying 35.186.224.24...
Connected to api.spotify.com.
Escape character is '^]'.
```

I just tested ncspot and Spot, and they both work fine.

I have not configured any proxy on my machine, but I'm guessing it has something to do with a specificity of my university's network. I'm not really sure how to investigate that though :(

**Comment by SO9010** [comment]
@TrakJohnson thanks for the comprehensive report. I just had an idea what could be the issue! Could you see if your uni WiFi allows you to get your location from this endpoint https://ip-api.com/ (you have to use your IP address). 

We need to do this in psst to set a region for the homepage to work. Maybe if this is the issue we need to have an option where the user can put in their region themself.

**Comment by Joedmin** [comment]
I'm on eduroam and do not have this issue. I use Fedora 42 with KDE

**Comment by TrakJohnson** [comment]
Sorry for the very long wait, I will try what you suggest as soon as I'm back on eduroam. Good to know that it doesn't affect many :)

**Comment by evs-xsarus** [comment]
@jacksongoode could this be implemented: https://developer.spotify.com/documentation/web-api/concepts/redirect_uri
